pkgs/development/tools/misc/blackmagic/default.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, lib, fetchFromGitHub
+, gcc-arm-embedded, bash, libftdi
+, python, pythonPackages
+}:
+
+with lib;
+
+stdenv.mkDerivation rec {
+  name = "blackmagic-${version}";
+  version = "1.6.1";
+
+  src = fetchFromGitHub {
+    owner = "blacksphere";
+    repo = "blackmagic";
+    rev = "d3a8f27fdbf952194e8fc5ce9b2fc9bcef7c545c";
+    sha256 = "0c3l7cfqag3g7zrfn4mmikkx7076hb1r856ybhhdh0f6zji2j6jx";
+    fetchSubmodules = true;
+  };
+
+  buildInputs = [
+    gcc-arm-embedded
+    libftdi
+    python
+    pythonPackages.intelhex
```

Comment:
It has one executable that can be used when you don't have the proper hardware.  When you do have the proper hardware the package includes the built firmware so you can flash your device.